spec/requests/providers/means/cash_outgoings_controller_spec.rb

```diff
@@ -0,0 +1,239 @@
+                   :with_applicant,
+                   :with_non_passported_state_machine,
+                   :applicant_entering_means,
+                   transaction_types: []
+          end
+
+          it "redirects to the has dependants page" do
+            request
+            expect(response).to redirect_to(providers_legal_aid_application_means_has_dependants_path(legal_aid_application))
+          end
+        end
+      end
+
+      context "with bank statement upload flow" do
+        before do
+          legal_aid_application.provider.permissions << Permission.find_or_create_by(role: "application.non_passported.bank_statement_upload.*")
+          legal_aid_application.update!(provider_received_citizen_consent: false)
+        end
+
+        context "with income and outgoing categories" do
+          let(:legal_aid_application) do
+            create(:legal_aid_application, :with_applicant,
+                   :with_non_passported_state_machine,
+                   :applicant_entering_means,
+                   transaction_types:)
+          end
+
+          let(:transaction_types) { [create(:transaction_type, :credit_with_standard_name), create(:transaction_type, :debit_with_standard_name)] }
+
+          it "redirects to has dependants page" do
+            request
+            expect(response).to redirect_to(providers_legal_aid_application_means_has_dependants_path(legal_aid_application))
+          end
+        end
+
+        context "without income or outgoing categories" do
+          let(:legal_aid_application) do
+            create(:legal_aid_application, :with_applicant,
+                   :with_non_passported_state_machine,
+                   :applicant_entering_means,
+                   transaction_types: [])
+          end
+
+          it "redirects to has dependants page" do
+            request
+            expect(response).to redirect_to(providers_legal_aid_application_means_has_dependants_path(legal_aid_application))
+          end
+        end
+      end
+    end
+
+    context "with nothing selected" do
+      let(:params) { nothing_selected_params }
+
+      it "updates the model attribute for no cash outgoings to true" do
+        expect { request }.to change { legal_aid_application.reload.no_cash_outgoings }.from(nil).to(true)
+      end
+    end
+
+    context "with invalid params" do
+      let(:params) { invalid_params }
+
+      before { request }
+
+      it "returns http success" do
+        expect(response).to have_http_status(:ok)
+      end
+
+      it "shows an error for no amount entered" do
+        expected_month = (Time.zone.today - 1.month).strftime("%B")
+        expect(response.body).to include(I18n.t("errors.aggregated_cash_outgoings.blank", category: "in maintenance", month: expected_month))
+      end
+
+      it "shows an error for an invalid amount" do
+        expect(response.body).to include(I18n.t("errors.aggregated_cash_outgoings.invalid_type"))
```

Comment:
expected error message is `invalid_type` which doesn't match the `it` block of invalid amount?

Reply:
done